text/0113-keyspace-level-gc.md

```diff
@@ -0,0 +1,68 @@
+
+So we propose the **Keyspace Level GC**:
+
+TiDB side:
+Isolate of GC safe point calculations between keyspaces (the concept is 'keyspace level GC').
+It will not affect other keyspaces GC safe point calculation.
+Keyspaces can be created by setting gc_management_type = keyspace_level_gc to enable keyspace level GC, then this keyspace can calculate GC safe point by itself.
+
+TiKV side:
+In GC process, it parses the keyspace id from the data key, use the keyspace meta config and the keyspace level GC safe point corresponding to the keyspace id to determine the GC safe point value of the data key and execute the GC logic.
+
+
+## Implementation in TiKV
+
+1. Get [keyspace meta](https://github.com/pingcap/kvproto/blob/d9297553c9009f569eaf4350f68a908f7811ee55/proto/keyspacepb.proto#L26C1-L33C2) and keyspace level GC safe point from PD:
+    - Create a keyspace meta watch service: Watch keyspace meta etcd path to get the keyspace meta and put it in the mapping from keyspace id to keyspace meta.
+    - Create a keyspace level gc watch service : Watch the etcd path of the keyspace GC safe point to get the GC safe point with keyspace level GC enabled, put it in the mapping from keyspace id to keyspace level gc.
+
+2. How to get GC safe point by mvcc key in Compaction Filter:
+ ![img.png](../media/keyspace-level-gc-get-gc-sp.png)
+
+3. How to determine if a keyspace uses a global GC safe point:
+ ![img.png](../media/keyspace-level-gc-is-global-gc.png)
+
+4. Use GC safe point to optimize trigger timing and assert judgment:
+   1. In the global GC, it will skip GC when GC safe point is 0 in create_compaction_filter.
+      After supporting keyspace level GC, GC is skipped if global GC safe point is 0 or if no keyspace level GC is initialized.
+   2. In the global GC, check_need_gc function return false in create_compaction_filter.
+      After supporting keyspace level GC, if props.min_ts > global GC safe point and props.min_ts > all keyspace level GC safe point will return false.
+   3. In the global GC, assert( safe_point > 0 ) when new compaction filter.
+      After supporting keyspace level GC, the assert condition is whether global GC safe point > 0 or any keyspace level GC safe point has been initialized.
+
+5. Support use keyspace level GC for data import and export:
+   1. When using BR, CDC, Lightning, Dumpling to import and export data specified keyspace, you need to update the service safe point for the specified keyspace. When the task starts, When the task starts, it needs to get keyspace meta first to determine whether to execute the keyspace level gc logic.
+
+## Upgrade from `global GC` to `keyspace level GC`
+It will be introduced in another RFC.
+
+
+## Global BR restore
```

Comment:
I think only backup has operations on GC safe point / service safe point, but not restore. Backup needs to check the backup ts >= GC safe point, and set backup ts as service safe point before backup is done.

```suggestion
## Global Backup
```